### PR TITLE
Database backup restore [RDO-214]

### PR DESCRIPTION
**Pull request by tomas-matus**
## What
- [x] change chart to use as source of database authentication information k8s secret
- [x] split chart to dirs backup, semaphore, postgresql
- [x] add database restore feature
- [x] change values.yml structure to database, semaphore, backup, restore "primary keys"

### CODE REVIEW

**Comment by tomas-matus** [comment]
@Teebor-Choka rebased, please take a look when possible.

File: charts/semaphore/Chart.yaml [hunk 1]
```diff
@@ -8,6 +8,6 @@ description: |
 
 type: application
 
-version: 0.1.2
+version: 0.1.3
```
Comment (PhotoTeeborChoka) [inline charts/semaphore/Chart.yaml]: You redefined the entire values.yml file, this is at least a major version update. Let's say we go to 1.0.0.

File: charts/semaphore/templates/_helpers.tpl [hunk 2]
```diff
@@ -66,72 +69,32 @@ app.kubernetes.io/name: {{ include "semaphore.name" . }}
...
-{{- define "semaphore.database.volume.size" -}}
-{{- default "5Gi" .Values.database.volumeSize }}
-{{- end }}
-
-
 {{/*
 Semaphore configuration secret resource name
 */}}
-{{- define "semaphore.config.secretName" -}}
+{{- define "semaphore.config.configMapName" -}}
```
Comment (PhotoTeeborChoka) [inline charts/semaphore/templates/_helpers.tpl]: The comment does not match the definition.

File: charts/semaphore/templates/backup/cronjob.yaml [hunk 3]
```diff
@@ -23,97 +23,93 @@ spec:
...
             - name: create-database-dump-file
-              image: "{{ .Values.database.image }}"
+              image: "{{ .Values.database.image.name }}:{{ .Values.database.image.tag }}"
               imagePullPolicy: "{{ .Values.database.imagePullPolicy }}"
               command: [ 'sh', '-c']
               args:
-                - set -exuv;
-                  echo "${POSTGRES_HOST}:${POSTGRES_PORT}:${POSTGRES_DATABASE}:${POSTGRES_USER}:${POSTGRES_PASSWORD}" > "$HOME/.pgpass";
+                - echo "${POSTGRES_HOST}:${POSTGRES_PORT}:${POSTGRES_DATABASE}:${POSTGRES_USER}:${POSTGRES_PASSWORD}" > "$HOME/.pgpass";
+                  set -exuv;
```
Comment (PhotoTeeborChoka) [inline charts/semaphore/templates/backup/cronjob.yaml:31]: Why did you swap the order?

File: charts/semaphore/templates/postgresql/persistent-volume-claim.yaml [hunk 4]
```diff
@@ -9,10 +9,10 @@ metadata:
   name: {{ include "semaphore.database.volume.name" . }}
 spec:
   accessModes:
-    - ReadWriteOnce
+    {{ toYaml .Values.database.persistence.accessModes | indent 4}}
   resources:
     requests:
-      storage: {{ include "semaphore.database.volume.size" . }}
+      storage: "{{ default "10Gi" .Values.database.persistence.volumeSize }}"
```
Comment (PhotoTeeborChoka) [inline charts/semaphore/templates/postgresql/persistent-volume-claim.yaml]: Let's have a default in values.yaml, and not here.

**Review by PhotoTeeborChoka — CHANGES_REQUESTED** [review]

File: charts/semaphore/Chart.yaml (hunk 1 above)
Comment (tomas-matus) [reply charts/semaphore/Chart.yaml]: changed

File: charts/semaphore/templates/_helpers.tpl (hunk 2 above)
Comment (tomas-matus) [reply charts/semaphore/templates/_helpers.tpl]: changed

File: charts/semaphore/templates/backup/cronjob.yaml (hunk 3 above)
Comment (tomas-matus) [reply charts/semaphore/templates/backup/cronjob.yaml:31]: to now show password in container logs

File: charts/semaphore/templates/postgresql/persistent-volume-claim.yaml (hunk 4 above)
Comment (tomas-matus) [reply charts/semaphore/templates/postgresql/persistent-volume-claim.yaml]: values.yaml have value, this was meant as fallback value

removed it

**Review by PhotoTeeborChoka — APPROVED** [review]

**Comment by PhotoTeeborChoka** [comment]
Please, deploy to test that everything works properly.

**Comment by tomas-matus** [comment]
> Please, deploy to test that everything works properly.

tested, working, just in case not merging, waiting for confirmation to merge

**Pull request merged**